uaclient/cli.py

```diff
@@ -1114,22 +1129,71 @@ def action_auto_attach(args, *, cfg):
 @assert_root
 @assert_lock_file("ua attach")
 def action_attach(args, *, cfg):
-    if not args.token:
+    if not args.token and not args.attach_config:
         raise exceptions.UserFacingError(
             ua_status.MESSAGE_ATTACH_REQUIRES_TOKEN
         )
-    try:
-        actions.attach_with_token(
-            cfg, token=args.token, allow_enable=args.auto_enable
+    if args.token and args.attach_config:
+        raise exceptions.UserFacingError(
+            ua_status.MESSAGE_ATTACH_TOKEN_ARG_XOR_CONFIG
         )
+
+    if args.token:
+        token = args.token
+        enable_services_override = None
+    else:
+        try:
+            attach_config = AttachActionsConfigFile.from_dict(
+                yaml.safe_load(args.attach_config)
+            )
+        except IncorrectTypeError as e:
+            raise exceptions.UserFacingError(
+                textwrap.fill(
+                    "Error while reading {}: {}".format(
+                        args.attach_config.name, e.msg
+                    ),
+                    width=PRINT_WRAP_WIDTH,
+                )
+            )
+        token = attach_config.token
+        enable_services_override = attach_config.enable_services
+
+    allow_enable = args.auto_enable and enable_services_override is None
+
+    try:
+        actions.attach_with_token(cfg, token=token, allow_enable=allow_enable)
     except util.UrlError:
         event.info(ua_status.MESSAGE_ATTACH_FAILURE)
         return 1
     except exceptions.UserFacingError:
         return 1
     else:
+        ret = 0
+        if enable_services_override is not None and args.auto_enable:
+            found, not_found = get_valid_entitlement_names(
+                enable_services_override
+            )
+            for name in not_found:
+                logging.warning(
+                    ua_status.MESSAGE_INVALID_SERVICE_OP_FAILURE_TMPL.format(  # noqa: E501
+                        operation="enable", name=name, service_msg=""
+                    )
+                )
```

Comment:
Can we just have one log with all the services that are not found ?

Reply:
Yep thats a great idea